site/en/blog/fugu-showcase/index.md

```diff
@@ -34,6 +36,59 @@ Easter egg, the Project Fugu API Showcase is of course
 <a href="https://tomayac.github.io/fugu-showcase/data/#tomayac.github.io!fugu-showcase!data" target="showcase">contained
 in the Project Fugu API Showcase</a>. Happy browsing!
 
-<div style="height: 100%; width: 100%;">
-  <iframe title="Fugu showcase" name="showcase" style="min-height: 800px; width: 100%; border: 0;" src="https://tomayac.github.io/fugu-showcase/data/" allow="web-share; clipboard-write; clipboard"></iframe>
+## Showcase
+
+<div class="showcase-container" style="height: 100%; width: 100%">
+  <noscript>
+    <iframe
+      title="Fugu showcase"
+      name="showcase"
+      style="min-height: 1000px; width: 100%; border: solid 1px var(--color-hairline);"
+      src="https://tomayac.github.io/fugu-showcase/data/"
+      allow="web-share; clipboard-write; clipboard"
+    ></iframe>
+  </noscript>
 </div>
+
+<script>
+  const SHOWCASE_URL = 'https://tomayac.github.io/fugu-showcase/data/';
+  
+  const iframe = document.createElement('iframe');
```

Comment:
What's the advantage of creating the `iframe` imperatively instead of just using the version created via HTML? Are you worried about a timing issue around when the `load` event listener is added?

If so, you could consider:

- Using the `onload` attribute in HTML to invoke a function that's defined before the HTML.
- Just unconditionally calling `iframe.contentWindow.postMessage()` and relying on the fact that `postMessage()` will be [queued](https://www.w3.org/TR/2021/NOTE-workers-20210128/#:~:text=workers%20has%20its-,port%20message%20queue,-implicitly%20enabled%20when) until the document being posted to has fully loaded.

It would be cleaner to avoid defining the `iframe`'s in both HTML (for `<noscript>`) and JavaScript.

Reply:
I used `<iframe onload>`, but the only way I found to make the `Window.postMessage()` call to work was via `frames[0]`, which sounds very brittle should we add more frames. The intuitive `this.contentWindow.postMessage()`  didn't work. So I came up with the present `<noscript>` and scripted approach, which works universally.

Reply:
Apologies, but I am missing some nuance here. Which `Window.postMessage()` call are you talking about? How does `frames[0]` factor in?

Reply:
I managed to get it working with the `onload` attribute. PTAL.